File: tswebapi/TSModel/Dominio/Consejeria/EntrevistaPostAborto.cs
using System;

namespace TSModel.Dominio.Consejeria
{
    public class EntrevistaPostAborto : EntidadBase
    {
        public virtual  Consejeria Consejeria { get; set; }

        public virtual  DateTime Fecha { get; set; }
        public virtual  string ProcedimientoObservaciones { get; set; }

        public virtual  bool ProcedimientoHecho { get; set; }
        public virtual  bool ProcedimientoNoContinua { get; set; }
        public virtual  bool ProcedimientoNoAbortoEspontaneo { get; set; }
        public virtual  string ProcedimientoNoOtro { get; set; }

        public virtual  bool ProcedimientoSiInformado { get; set; }
        public virtual  bool ProcedimientoSiOtra { get; set; }
        public virtual  bool ProcedimientoSiViaV { get; set; }
        public virtual  bool ProcedimientoSiViaSL { get; set; }
        public virtual  string ProcedimientoSiOtro { get; set; }

        public virtual  bool AccedioPorFarmacia { get; set; }
        public virtual  bool AccedioPorConocido { get; set; }
        public virtual  bool AccedioPorInternet { get; set; }
        public virtual  bool AccedioPorOrgSocial { get; set; }

        public virtual  bool PresentacionSuelto { get; set; }
        public virtual  bool PresentacionCaja20 { get; set; }
        public virtual  bool PresentacionCaja16 { get; set; }

        public virtual  bool EfectoAdversoNo { get; set; }
        public virtual  bool EfectoAdversoGastro { get; set; }
        public virtual  bool EfectoAdversoTemperatura { get; set; }
        public virtual  bool EfectoAdversoCafalea { get; set; }
        public virtual  string EfectoAdversoOtro { get; set; }

        public virtual  bool ComplicacionNo { get; set; }
        public virtual  bool ComplicacionHemorragia { get; set; }
        public virtual  bool ComplicacionInfeccion { get; set; }
        public virtual  string ComplicacionOtro { get; set; }

        public virtual  bool IndicacionGammaglobulina { get; set; }

        public virtual  DateTime EcografiaPostFecha { get; set; }
        public virtual  bool EcografiaPostNoRealizo { get; set; }
        public virtual  bool EcografiaPostAbortoCompleto { get; set; }
        public virtual  bool EcografiaPostHMR { get; set; }
        public virtual  bool EcografiaPostAbortoIncompleto { get; set; }
        public virtual  bool EcografiaPostEmbrionViable { get; set; }
        public virtual  bool EcografiaPostNuevaConsejeria { get; set; }
        public virtual  bool EcografiaPostDerivacion2Nivel { get; set; }
        public virtual  bool EcografiaPostConductaExpectante { get; set; }

        public virtual  bool ConcejeriaMACNo { get; set; }
        public virtual  bool ConcejeriaMACACO { get; set; }
        public virtual  bool ConcejeriaMACACI { get; set; }
        public virtual  bool ConcejeriaMACDIU { get; set; }
        public virtual  bool ConcejeriaMACPreservativo { get; set; }
        public virtual  bool ConcejeriaMACImplanteHormonal { get; set; }
    }
}

File: Angular7/myapp1/src/server/models/entrevistaPostAborto.js
const mongoose = require('mongoose');
const uniqueValidator = require('mongoose-unique-validator')

let Shcema = mongoose.Schema;
let entrevistaPostAbortoSchema = new Shcema({
    fecha:{
        type:Date,
        required: [true, 'La fecha es requerida']
    },
    procedimientoObservaciones:{
       type:String
    },
    procedimientoHecho:{
        type:Boolean,
        default:false
    },
    procedimientoNoContinua:{
        type:Boolean,
        default:false
    },
    procedimientoNoAbortoEspontaneo:{
        type:Boolean,
        default:false
    },
    procedimientoNoOtro:{
        type:String,
        required: [true, 'El nombre es requerido']
    },
    procedimientoNoOtro:{
        type:Boolean,
        default:false
    },
    procedimientoSiHecho:{
        type:Boolean,
        default:false
    },
    procedimientoSiInformado:{
        type:Boolean,
        default:false
    },
    procedimientoSiOtra:{
        type:Boolean,
        default:false
    },
    procedimientoSiViaV:{
        type:Boolean,
        default:false
    },
    procedimientoSiViaSL:{
        type:Boolean,
        default:false
    },
    procedimientoSiOtro:{
        type:Boolean,
        default:false
    },
    accedioPorFarmacia:{
        type:Boolean,
        default:false
    },
    accedioPorConocido:{
        type:Boolean,
        default:false
    },
    accedioPorInternet:{
        type:Boolean,
        default:false
    },
    pccedioPorOrgSocial:{
        type:Boolean,
        default:false
    },
    presentacionSuelto:{
        type:Boolean,
        default:false
    },
    presentacionCaja20:{
        type:Boolean,
        default:false
    },
    presentacionCaja16:{
        type:Boolean,
        default:false
    },
    efectoAdversoNo:{
        type:Boolean,
        default:false
    },
    efectoAdversoGastro:{
        type:Boolean,
        default:false
    },
    efectoAdversoTemperatura:{
        type:Boolean,
        default:false
    },
    efectoAdversoCafalea:{
        type:Boolean,
        default:false
    },
    efectoAdversoOtro:{
        type:String
    },
    complicacionNo:{
        type:Boolean,
        default:false
    },
    complicacionHemorragia:{
        type:Boolean,
        default:false
    },
    complicacionInfeccion:{
        type:Boolean,
        default:false
    },
    complicacionOtro:{
        type:String
    },
    indicacionGammaglobulina:{
        type:Boolean,
        default:false
    },
    ecografiaPostFecha:{
        type:Date
    },
    ecografiaPostNoRealizo:{
        type:Boolean,
        default:false
    },
    ecografiaPostAbortoCompleto:{
        type:Boolean,
        default:false
    },
    ecografiaPostAbortoIncompleto:{
        type:Boolean,
        default:false
    },
    ecografiaPostAbortoHMyR:{
        type:Boolean,
        default:false
    },
    ecografiaPostEmbrionViable:{
        type:Boolean,
        default:false
    },
    ecografiaPostNuevaConsejeria:{
        type:Boolean,
        default:false
    },
    ecografiaPostDerivacion2Nivel:{
        type:Boolean,
        default:false
    },
    ecografiaPostConductaExpectante:{
        type:Boolean,
        default:false
    },
    consejeriaMACNo:{
        type:Boolean,
        default:false
    },
    consejeriaMACACO:{
        type:Boolean,
        default:false
    },
    consejeriaMACACI:{
        type:Boolean,
        default:false
    },
    consejeriaMACDIU:{
        type:Boolean,
        default:false
    },
    consejeriaMACPreservativo:{
        type:Boolean,
        default:false
    },
    consejeriaMACImplanteHormonal:{
        type:Boolean,
        default:false
    },
    consejeriaMACACOLactancia:{
        type:Boolean,
        default:false
    },
    consejeriaId:{
        type:String
    }
});

entrevistaPostAbortoSchema.methods.toJSON = function (){
    let user = this;
    let userObject = user.toObject();
    delete userObject.password;
    return userObject;
}

entrevistaPostAbortoSchema.plugin(uniqueValidator, {message:'{PATH} debe de ser unico'});

module.exports = mongoose.model('Usuarias', entrevistaPostAbortoSchema);
